simtools/simulator.py

```diff
@@ -313,42 +326,55 @@ def _set_simulation_runner(self):
         Set simulation runners
 
         """
+        common_args = {
+            "label": self.label,
+            "simtel_source_path": self._simulator_source_path,
+        }
+        corsika_args = {
+            "mongo_db_config": self._mongo_db_config,
+            "site": self.site,
+            "layout_name": self.layout_name,
+            "corsika_parameters_file": self._corsika_parameters_file,
+            "corsika_config_data": self._corsika_config_data,
+        }
+        # TODO: This is not very elegant, find a nicer solution?
+        if self.simulator in ["simtel", "corsika_simtel"]:
+            simtel_args = {
+                "array_model": self.array_model,
+                "config_data": {
+                    "simtel_data_directory": self.config.data_directory,
+                    "primary": self.config.primary,
+                    "zenith_angle": self.config.zenith_angle * u.deg,
+                    "azimuth_angle": self.config.azimuth_angle * u.deg,
+                },
+            }
+        if self.simulator == "corsika":
+            self._set_corsika_runner(common_args | corsika_args)
         if self.simulator == "simtel":
-            self._set_simtel_runner()
-        elif self.simulator == "corsika":
-            self._set_corsika_runner()
+            self._set_simtel_runner(common_args | simtel_args)
+        if self.simulator == "corsika_simtel":
+            self._set_corsika_simtel_runner(common_args, corsika_args, simtel_args)
 
-    def _set_corsika_runner(self):
+    def _set_corsika_runner(self, simulator_args):
         """
         Creating CorsikaRunner.
 
         """
-        self._simulation_runner = CorsikaRunner(
-            mongo_db_config=self._mongo_db_config,
-            label=self.label,
-            site=self.site,
-            layout_name=self.layout_name,
-            simtel_source_path=self._simulator_source_path,
-            corsika_parameters_file=self._corsika_parameters_file,
-            corsika_config_data=self._corsika_config_data,
-        )
+        self._simulation_runner = CorsikaRunner(**simulator_args)
 
-    def _set_simtel_runner(self):
+    def _set_simtel_runner(self, simulator_args):
         """
         Creating a SimtelRunnerArray.
 
         """
-        self._simulation_runner = SimtelRunnerArray(
-            label=self.label,
-            array_model=self.array_model,
-            simtel_source_path=self._simulator_source_path,
-            config_data={
-                "simtel_data_directory": self.config.data_directory,
-                "primary": self.config.primary,
-                "zenith_angle": self.config.zenith_angle * u.deg,
-                "azimuth_angle": self.config.azimuth_angle * u.deg,
-            },
-        )
+        self._simulation_runner = SimtelRunnerArray(**simulator_args)
+
+    def _set_corsika_simtel_runner(self, common_args, corsika_args, simtel_args):
```

Comment:
I know it is a bit annoying, but it would be nice if we document also the internal functions for ourselves (the developers).

Reply:
This would be a lot of work to do consistently everywhere. Maybe in the next "maintenance period"?